Tests/NIOTests/ChannelTests.swift

```diff
@@ -2200,4 +2200,98 @@ public class ChannelTests: XCTestCase {
         XCTAssertNoThrow(try allDone.futureResult.wait())
         XCTAssertNoThrow(try sc.syncCloseAcceptingAlreadyClosed())
     }
+
+    func testConnectWithECONNRESETGetsTheRightError() throws {
+        let group = MultiThreadedEventLoopGroup(numThreads: 1)
+        defer {
+            XCTAssertNoThrow(try group.syncShutdownGracefully())
+        }
+        let serverSockSyncQ = DispatchQueue(label: "serverSock syncQ")
+        let serverSock = try serverSockSyncQ.sync { () -> Socket in
+            let sock = try Socket(protocolFamily: PF_INET, type: Posix.SOCK_STREAM)
+            XCTAssertNoThrow(try sock.bind(to: SocketAddress(ipAddress: "127.0.0.1", port: 0)))
+            XCTAssertNoThrow(try sock.withUnsafeFileDescriptor { fd in
+                try Posix.listen(descriptor: fd, backlog: 1)
+            })
+            return sock
+        }
+        let serverSockAddress = serverSockSyncQ.sync { try! serverSock.localAddress() }
+        defer {
+            XCTAssertNoThrow(try serverSockSyncQ.sync { try serverSock.close() })
+        }
+        let g = DispatchGroup()
+        let q = DispatchQueue(label: "q")
+        do {
+            // spawn 64 connects, likely some will already get ECONNREFUSED as the backlog isn't large enough
+            let fs = (0..<64).map { _ in ClientBootstrap(group: group).connect(to: serverSockAddress) }
+            q.async(group: g) { () -> Void in
+                while true {
+                    do {
+                        // try to accept one client
+                        let accepted = try serverSockSyncQ.sync { () -> CInt? in
+                            try serverSock.withUnsafeFileDescriptor { fd in
+                                var pfd = pollfd(fd: fd, events: Int16(POLLIN | POLLERR | POLLHUP), revents: 0)
+                                // wait a maximum of 100ms to accept a new client, just to be guaranteed we don't get stuck here.
+                                if poll(&pfd, 1, 100) == 1 {
+                                    return accept(fd, nil, nil)
+                                } else {
+                                    return nil
+                                }
+                            }
+                        }
+                        if let accepted = accepted {
+                            // set SO_LINGER to 0, which will cause most TCP stacks to send an RST packet
+                            var lin = linger(l_onoff: 1, l_linger: 0)
+                            setsockopt(accepted, SOL_SOCKET, SO_LINGER, &lin, socklen_t(MemoryLayout.size(ofValue: lin)))
```

Comment:
If `SO_LINGER` can get a `RST` packet, why do we need 64 connections?

Reply:
@Lukasa the `SO_LINGER` works well for Darwin, the 64 connections work well for Linux ;)

Reply:
Ah, can you update the comments to make that clear?